stdlib/public/Observation/Sources/Observation/CMakeLists.txt

```diff
@@ -10,42 +10,25 @@
 #
 #===----------------------------------------------------------------------===#
 
-# Linux and windows hosts are disabled for now; it is dependent upon earlyswiftsyntax
-# which does not build in the same way just yet
-if (${SWIFT_HOST_VARIANT_SDK} IN_LIST SWIFT_DARWIN_PLATFORMS)
-  list(APPEND swift_runtime_library_compile_flags -I${SWIFT_SOURCE_DIR}/stdlib/include -I${SWIFT_SOURCE_DIR}/include)
-
-  set(SWIFT_OBSERVATION_SWIFT_FLAGS)
-  set(SWIFT_OBSERVATION_DEPENDS)
-
-  list(APPEND SWIFT_OBSERVATION_SWIFT_FLAGS
-    "-enable-experimental-feature" "Macros"
-    "-DSWIFT_OBSERVATION_MACROS"
```

Comment:
We're no longer adding `-DSWIFT_OBSERVATION_MACROS` but the library depends on it, ie.
```
#if SWIFT_OBSERVATION_MACROS
...
#end
```
in `stdlib/public/Observation/Sources/Observation/Macros.swift`. I don't think that check is actually needed, can we just remove it?